server/services/report/draftReportService.ts

```diff
@@ -27,8 +27,14 @@ export default class DraftReportService {
     return getReportStatus(report)
   }
 
-  public update(params: UpdateParams): Promise<number> {
-    return this.updateDraftReport.update(params)
+  public process(
+    currentUser: LoggedInUser,
+    bookingId: number,
+    formName: string,
+    updatedSection: unknown,
+    incidentDate?: Date | null
+  ): Promise<void> {
+    return this.updateDraftReport.process(currentUser, bookingId, formName, updatedSection, incidentDate)
```

Comment:
this just delegates down - not sure if there is a way of defining the args as a type so don't have to duplicate, something like: 


public process(
    currentUser: LoggedInUser,
    bookingId: number,
    formName: string,
    updatedSection: unknown,
    incidentDate?: Date | null
  ): Promise<void> {
    return this.updateDraftReport.process(currentUser, bookingId, formName, updatedSection, incidentDate)
 

 
this just delegates down - not sure if there is a way of defining the args as a type so don't have to duplicate, something like a typesafe version of: 
```ts
  public process(
    ...args: UpdateParams
  ): Promise<void> {
    return this.updateDraftReport.process(...args)
}
```

Reply:
might be able to do this in [typescript 3 actually](https://auth0.com/blog/typescript-3-exploring-tuples-the-unknown-type/#:~:text=A%20tuple%20is%20a%20TypeScript,need%20not%20be%20the%20same).